nixos/modules/services/development/lorri.nix

```diff
@@ -0,0 +1,57 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.lorri;
+  socketUnit = "lorri";
+  socketPath = "lorri/daemon.socket";
+in with lib; {
+  options = {
+    services.lorri = {
+      enable = mkOption {
+        default = false;
+        type = types.bool;
+        description = ''
+          This option enables a systemd socket unit that listens on the
+          well-known address used by lorri, a nix-shell replacement for project
+          development. When a lorri client connects to the socket, systemd
+          starts the lorri daemon as a service.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.user.sockets.${socketUnit} = {
```

Comment:
@aszlig can you take a quick look at this socket definition? Anything wrong with the socket-activation part? Possible improvements?

Reply:
I am not aware of a restriction on the systemd side that would force the socket and the service to have the same name. I used a variable here to "bind" everything that _must_ use the same identifier together.

Reply:
from https://www.freedesktop.org/software/systemd/man/systemd.socket.html#Service= :
```
Service=¶

    Specifies the service unit name to activate on incoming traffic. This setting is only allowed for sockets with Accept=no. It defaults to the service that bears the same name as the socket (with the suffix replaced). In most cases, it should not be necessary to use this option. Note that setting this parameter might result in additional dependencies to be added to the unit (see above).
```

In our case, this currently just works by accident because we called the service `lorri.service`.

Again, I'd really suggest to just hardcode lorri.socket and lorri.daemon, it makes things more readable too, and there's no way to override anyways.

Reply:
Thanks! I'll make the change.